parquet/src/main/java/org/apache/iceberg/parquet/MessageTypeToType.java

```diff
@@ -56,36 +60,38 @@
 
   @Override
   public Type message(MessageType message, List<Type> fields) {
-    return struct(message, fields);
+    Type struct = struct(message, fields);
+    return struct != null ? struct : Types.StructType.of(Lists.newArrayList());
```

Comment:
This part requires attention. Tests for empty projections would fail otherwise.

Reply:
Tests for empty projections?

Reply:
For example, `testEmptyProjection` in `TestReadProjection`.